src/riak_core_job_sup.erl

```diff
@@ -0,0 +1,208 @@
+%% -------------------------------------------------------------------
+%%
+%% Copyright (c) 2016 Basho Technologies, Inc.
+%%
+%% This file is provided to you under the Apache License,
+%% Version 2.0 (the "License"); you may not use this file
+%% except in compliance with the License.  You may obtain
+%% a copy of the License at
+%%
+%%   http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing,
+%% software distributed under the License is distributed on an
+%% "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+%% KIND, either express or implied.  See the License for the
+%% specific language governing permissions and limitations
+%% under the License.
+%%
+%% -------------------------------------------------------------------
+
+%% @private
+%% @doc Internal Job Runner Supervisor.
+%%
+%% Refer to the {@link riak_core_job_service} module for the process model.
+%%
+-module(riak_core_job_sup).
+-behaviour(supervisor).
+
+% Private API
+-export([
+    start_runner/0,
+    stop_runner/1
+]).
+
+% Supervisor API
+-export([
+    init/1,
+    start_link/0
+]).
+
+-ifdef(TEST).
+-export([
+    start_test_sup/0,
+    stop_test_sup/1
+]).
+-endif.
+
+-include("riak_core_job_internal.hrl").
+-include("riak_core_sup_internal.hrl").
+
+-define(RUNNER_SUP,
+    {{simple_one_for_one, 30, 60}, [
+        {job_runner,
+            {riak_core_job_runner, start_link, [?job_run_ctl_token]},
+            temporary, ?WORK_RUN_SHUTDOWN_TIMEOUT,
+            worker, [riak_core_job_runner]} ]}).
+
+%% ===================================================================
+%% Private API
+%% ===================================================================
+
+-spec start_runner() -> {ok, pid()} | {error, term()}.
+%% @private
+%% @doc Start a new job runner process.
+%%
+start_runner() ->
+    supervisor:start_child(?MODULE, []).
+
+-spec stop_runner(Runner :: pid()) -> ok | {error, term()}.
+%% @private
+%% @doc Stop a job runner process.
+%%
+stop_runner(Runner) when erlang:is_pid(Runner) ->
+    case supervisor:terminate_child(?MODULE, Runner) of
+        ok ->
+            ok;
+        {error, not_found} = Error ->
+            % Errr, is it a runner that might have been started by a previous
```

Comment:
If the supervisor died, any runners running underneath it should die too, since they are linked. This extra cleanup code is unnecessary, and it may cause delays in the execution of this function since the `receive` block will always time out.